webhooks-extension/endpoints/credentials.go

```diff
@@ -0,0 +1,261 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+		http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoints
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"time"
+
+	restful "github.com/emicklei/go-restful"
+	logging "github.com/tektoncd/experimental/webhooks-extension/pkg/logging"
+	"github.com/tektoncd/experimental/webhooks-extension/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// 'credentials' from the webhooks-extension's point of view, are access tokens. That's the only sort we handle right now.
+type credential struct {
+	Name            string `json:"name"`
+	AccessToken     string `json:"accesstoken"`
+	SecretToken     string `json:"secrettoken,omitempty"`
+	ResourceVersion string `json:"resourceversion,omitempty"`
+}
+
+func (r Resource) createCredential(request *restful.Request, response *restful.Response) {
+	cred := credential{}
+	if err := getQueryEntity(&cred, request, response); err != nil {
+		return
+	}
+	if !r.verifyCredentialParameters(cred, response) {
+		return
+	}
+	targetNamespace := request.QueryParameter("namespace")
+	if targetNamespace != "" {
+		targetNamespace := r.Defaults.Namespace
+		if targetNamespace == "" {
+			targetNamespace = "default"
+		}
+	}
+	if !r.verifyNamespaceExists(targetNamespace, response) {
+		return
+	}
+	secret := credentialToSecret(cred, targetNamespace, response)
+	if _, err := r.K8sClient.CoreV1().Secrets(targetNamespace).Create(secret); err != nil {
+		errorMessage := fmt.Sprintf("Error creating secret in K8sClient: %s", err.Error())
+		utils.RespondMessageAndLogError(response, err, errorMessage, http.StatusBadRequest)
+		return
+	}
+	writeResponseLocation(request, response, cred.Name)
+}
+
+/* Delete a Secret by name.
+ * Required path parameters:
+ *  - Namespace
+ *  - Name
+ */
+func (r Resource) deleteCredential(request *restful.Request, response *restful.Response) {
+	targetNamespace := request.QueryParameter("namespace")
+	if targetNamespace != "" {
+		targetNamespace := r.Defaults.Namespace
+		if targetNamespace == "" {
+			targetNamespace = "default"
+		}
+	}
+	if !r.verifyNamespaceExists(targetNamespace, response) {
+		return
+	}
+	credName := request.PathParameter("name")
+	if !r.verifySecretExists(credName, targetNamespace, response) {
+		return
+	}
+	_, err := r.K8sClient.CoreV1().Secrets(targetNamespace).Get(credName, metav1.GetOptions{})
```

Comment:
I'm wondering if we need to check the revision ..... someone might delete and recreate with the same name before i press the delete button in the ui?  Might be slight overkill right now and would mean the front end needs to pass the revision.  Open to discussion on this one.

Additionally....

1. why would we do this after you have just done a verifySecretExists? surely we are just repeating ourselves.
2. If you had a reason to do it, call the secretExists function

Reply:
http DELETE doesn't take a body so we'd have to pass the revision as a query param. Shall we have that conversation in dashboard first, and then add ?revision here if we also decide to do it in dashboard?